include/mxnet/tuple.h

```diff
@@ -220,6 +235,10 @@ class Tuple {
    * \return the ostream
    */
   friend std::ostream &operator<<(std::ostream &os, const Tuple<ValueType> &t) {
+    if (t.ndim() == -1) {
```

Comment:
should there be an explicit method for checking whether shape is known?

Reply:
By the current convention, it's defined in `operator_common.h` as `shape_is_known`, previously known as `shape_is_none`. For this case specifically, we should not use that function, because we want to just check `ndim`. For this release, I suggest we not move it to keep the convention as is. For the next major release, we can move the util function closer to where the object is defined.